frankenstein/distributor.go

```diff
@@ -195,3 +245,28 @@ func (d *Distributor) LabelValuesForLabelName(ctx context.Context, labelName mod
 func (*Distributor) NeedsThrottling(_ context.Context) bool {
 	return false
 }
+
+// Describe implements prometheus.Collector.
+func (d *Distributor) Describe(ch chan<- *prometheus.Desc) {
+	d.queryDuration.Describe(ch)
+	ch <- d.consulUpdates.Desc()
+	ch <- d.receivedSamples.Desc()
+	d.sendDuration.Describe(ch)
+	ch <- numClientsDesc
+}
+
+// Collect implements prometheus.Collector.
+func (d *Distributor) Collect(ch chan<- prometheus.Metric) {
+	d.queryDuration.Collect(ch)
+	ch <- d.consulUpdates
+	ch <- d.receivedSamples
+	d.sendDuration.Collect(ch)
+
+	d.clientsMtx.RLock()
+	defer d.clientsMtx.RUnlock()
+	ch <- prometheus.MustNewConstMetric(
+		numClientsDesc,
+		prometheus.GaugeValue,
+		float64(len(d.clients)),
+	)
+}
```

Comment:
To be able to set the value of a metric at scrape time, you usually implement the `Collector` interface, and once you do that for one metric, it seems cleaner to me to also encapsulate all the other metrics within the object they belong to instead of registering both the Distributor as a collector, and also some of its metrics individually.

Ok, I could also have used `GaugeFunc` for the one callback metric (actually didn't think of that), but I also generally like it when a type doesn't just register global metrics for you, but offers you an interface for doing it yourself. It doesn't currently matter for Frankenstein, but if you are writing a package that is supposed to be used by others, they might want to have control over how metrics are registered.